SwiftCompilerSources/Sources/Optimizer/DataStructures/InstructionRange.swift

```diff
@@ -159,3 +168,17 @@ struct InstructionRange : CustomStringConvertible, NoReflectionChildren {
     blockRange.deinitialize()
   }
 }
+
+extension InstructionRange {
+  // TODO: Add direct support for phis or function args. This workaround will be problematic when used in a pass that mutates the block.
+  static func getRepresentativeInstruction(for value: Value) -> Instruction {
```

Comment:
This is a terrible hack because `InstructionRange` doesn't support ranges that start at block boundaries, and I just don't have time to design that right now. It's incorrect to say the a block argument begins at the first instruction. The first instruction should instead be seen as within the range and we should be able to insert instructions above it.

 (Note that we also need some kind of MultiInstructionRange that supports multiple `begin` values)

Reply:
I converted `getRepresentativeInstruction` to an initializer since we don't plan on modeling instruction ranges that start at block boundaries.